### PR TITLE
dreprecated PageRouteBuilder.pageBuilder parameter

### PR DESCRIPTION
**Pull request by chunhtai**
## Description
The pageBuilder is confusing with the new page api. This pr deprecates the pagebuilder and rename it to bodyBuilder

page api pr https://github.com/flutter/flutter/pull/50362/files

## Related Issues

https://github.com/flutter/flutter/issues/45938

## Tests

I added the following tests:

Updates existing tests

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/pages.dart [hunk 1]
```diff
@@ -2,6 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// TODO(chunhtai): remove ignoring deprecated member use analysis
+// when PageRouteBuilder.pageBuilder parameter is removed. See
+// ignore_for_file: deprecated_member_use_from_same_package
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/pages.dart]: please use explicit //ignores everywhere so we don't accidentally ignore unrelated things

File: packages/flutter/lib/src/widgets/pages.dart [hunk 2]
```diff
@@ -42,16 +46,21 @@ Widget _defaultTransitionsBuilder(BuildContext context, Animation<double> animat
...
   /// Creates a route that delegates to builder callbacks.
   ///
-  /// The [pageBuilder], [transitionsBuilder], [opaque], [barrierDismissible],
+  /// The [bodyBuilder], [transitionsBuilder], [opaque], [barrierDismissible],
   /// [maintainState], and [fullscreenDialog] arguments must not be null.
   PageRouteBuilder({
     RouteSettings settings,
-    @required this.pageBuilder,
+    @Deprecated(
+      'Use bodyBuilder instead.'
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/pages.dart]: missing trailing space in string

File: packages/flutter/lib/src/widgets/pages.dart [hunk 3]
```diff
@@ -42,16 +46,21 @@ Widget _defaultTransitionsBuilder(BuildContext context, Animation<double> animat
...
+      'This feature was deprecated after v1.12.13'
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/pages.dart]: missing trailing period in string

File: packages/flutter/lib/src/widgets/pages.dart [hunk 4]
```diff
@@ -60,18 +69,19 @@ class PageRouteBuilder<T> extends PageRoute<T> {
     this.barrierLabel,
     this.maintainState = true,
     bool fullscreenDialog = false,
-  }) : assert(pageBuilder != null),
+  }) : assert((bodyBuilder ?? pageBuilder) != null),
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/pages.dart]: this wouldn't catch both of them being non-null

you want something like
```
  assert(bodyBuilder == null || pageBuilder == null),
  assert(bodyBuilder != null || pageBuilder != null),
```
...or:
```
  assert((bodyBuilder == null) != (pageBuilder == null)),
```

**Comment by Hixie** [comment]
LGTM modulo test failures and comments

File: packages/flutter/lib/src/widgets/pages.dart (hunk 4 above)
Comment (chunhtai) [reply packages/flutter/lib/src/widgets/pages.dart]: Do we want to block both of them being non-null? it will prefer bodyBuilder in this case.

**Comment by chunhtai** [comment]
Closing this pr as we don't want to take this approach

**Pull request closed without merging**